### PR TITLE
Expose /metrics endpoint in Prometheus format

### LINKED ISSUE #29: Expose prometheus metrics
It would be helpful if we expose `/metrics` with prometheus metrics
this lib looks like a good fit: https://github.com/deadtrickster/prometheus.erl

In the metrics, we can collect (not limited to)
- https response time buckets (one metrics for each api)
- consumer group lagging info

### PR DESCRIPTION
**Pull request by andrusha**
Closes #29. My first Elixir contribution, would appreciate feedback.

## Todo
- [x] Remove duplication (for each consumer group) of `topic_offset`
- [x] Tests
- [x]  Manual testing instructions
- [x] Verify on dockerized Prometheus

## Notes
* Only text format is available. 
* No client library is used, proved to be overly complex, contributions welcome.

## Output
```
kafka_consumer_group_offset{consumer_group="cat", topic="disputes", partition="0"} 58
kafka_consumer_group_offset{consumer_group="dog", topic="disputes", partition="0"} 44
kafka_consumer_group_offset{consumer_group="mew", topic="disputes", partition="0"} 48
kafka_consumer_group_offset{consumer_group="mew", topic="something", partition="0"} 3
kafka_topic_offset{topic="disputes" partition="0"} 58
kafka_topic_offset{topic="something" partition="0"} 14
```

## Docs
https://prometheus.io/docs/instrumenting/exposition_formats/#format-version-0-0-4
https://prometheus.io/docs/concepts/data_model/
https://prometheus.io/docs/practices/naming/

### CODE REVIEW

File: web/controllers/metrics_controller.ex [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    Entrypoint for Prometheus metrics collector.
+
+    Only topics with consumer groups are reported.
+  """
+  require Logger
+  use Kastlex.Web, :controller
+
+  def fetch(conn, _params) do
+    metrics = Enum.join(offsets(), "\n")
+    send_resp(conn, 200, metrics)
```
Comment (id) [inline web/controllers/metrics_controller.ex]: I think we're missing content type here https://prometheus.io/docs/instrumenting/exposition_formats/#format-version-0-0-4
```
   conn
   |> put_resp_content_type("text/plain; version=0.0.4")
   |> send_resp(200, metrics)
```

**Comment by id** [comment]
I see only one good way of removing duplication of `topic_offset`: first send all topic offsets, then send status for all consumer groups.

**Comment by andrusha** [comment]
@id if I do `start-test-broker.sh` and `mix test` afterwards I'm getting a bunch of error connections, although I can see container running https://gist.github.com/andrusha/48f0633545a1681f7f30437092ac8b92

What would be the right way to bring up environment and test this piece?

**Comment by zmstone** [comment]
why don't use https://github.com/deadtrickster/prometheus.erl
or https://github.com/deadtrickster/prometheus.ex

**Comment by andrusha** [comment]
@zmstone I couldn't figure out how to fit it easily with our use-case, mind that I have 0 Elixir experience too. We could pair up on the task if you're up to it.

**Comment by id** [comment]
@zmstone we tried to figure out together with @andrusha how to integrate it, our conclusion was formatting metrics for prometheus manually is just way simpler.

**Comment by zmstone** [comment]
It is perhaps trivial to generated the text lines for prometheus because the protocol is simple enough because all the values (offsets as gauge?) we need is already in place.
But if we want to emit other types, like counters and histograms, we'll have to implement something more than just a formatter.

Or maybe the formatter is not trivial either, take a at the escaping logic: https://github.com/deadtrickster/prometheus.erl/blob/master/src/formats/prometheus_text_format.erl)

**Comment by zmstone** [comment]
Missing `# TYPE` annotation ?

**Comment by andrusha** [comment]
@zmstone YAGNI, for the use-case of monitoring offsets on Kafka I think we're well covered.

> Or maybe the formatter is not trivial either, take a at the escaping logic
We luckied out here both kafka consumer group id and topic have to conform to a specific format, which doesn't need to be escaped:
https://github.com/apache/kafka/blob/0.10.2/core/src/main/scala/kafka/common/Topic.scala#L29-L31
https://github.com/apache/kafka/blob/f85c18032be5d583c2ca8f9fa1eb1b5ec55e59fa/core/src/main/scala/kafka/consumer/ConsumerConfig.scala#L72-L74
https://github.com/apache/kafka/blob/41e676d29587042994a72baa5000a8861a075c8c/core/src/main/scala/kafka/common/Config.scala#L25-L28

> Missing # TYPE annotation ?
By their spec it'll default to `untyped`. Do you think type is needed?

**Comment by id** [comment]
Verified on local prometheus instance.

**Pull request merged**